src/browser.ts

```diff
@@ -59,6 +60,17 @@ export abstract class BrowserBase extends EventEmitter implements Browser {
     download._reportFinished(error);
     this._downloads.delete(uuid);
   }
+
+  async close() {
+    const disconnectEventPromise = new Promise(x => this.once(Events.Browser.Disconnected, x));
+    if (this._ownedServer) {
+      await this._ownedServer.close();
+    } else {
+      await Promise.all(this.contexts().map(context => context.close()));
+      this._disconnect();
+    }
+    await disconnectEventPromise;
```

Comment:
You might only wait for this if you were not connected.

Reply:
Done.